### ISSUE TITLE
NUTS won't sample in Hierarchical multinomial

### ISSUE THREAD

**Issue by gcgibson** [pymc-devs/pymc#3166]
## Description of your problem
I am trying to get a relatively simply Hierarchical Multinomial example working. Here I have attached a small dummy example that  never seems to be able to start sampling. The basic idea is to have a population level draw of multinomial probabilities from a Dirichlet, and then group level probabilities given by a logit Normal with mean centered around population probabilities. Please let me know if you have any questions!

**Please provide a minimal, self-contained, and reproducible example.**
```python
  D=26
  true_p = np.random.dirichlet(np.ones(D))
  reporting_triangles = []
  
  for i in range(100):
	reporting_triangles.append(np.random.multinomial(100,true_p))

  
  tmp = np.array(reporting_triangles)
  tmp = tmp.reshape((5,20,D))  

  s1 = tmp.shape[0]
  s2 = tmp.shape[1]
  
  province_ids = []
  for i in range(s2):
    for j in range(s1):
      province_ids.append(i)
      

 
  observed_data = tmp.reshape((s1*s2,D))
  n_provinces = len(province_ids)

  with pm.Model() as hierarchical_model:
    # Hyperpriors for group nodes
    mu_a = pm.Dirichlet('alpha', a=np.ones(D),shape=D)
    
    mu_a_inv = pm.math.invlogit(mu_a)
    
    
    a = pm.MvNormal('a', mu=mu_a_inv, cov=np.eye(D), shape=(n_provinces,D))
   

    radon_est = pm.math.logit(a[province_ids])

    # Data likelihood
    radon_like = pm.Multinomial('radon_like', n=observed_data.sum(axis=1), p=radon_est, observed=observed_data)
  
  with hierarchical_model:
    hierarchical_trace = pm.sample()
```

**Please provide the full traceback.**
```python
Auto-assigning NUTS sampler...
Initializing NUTS using jitter+adapt_diag...
Multiprocess sampling (2 chains in 2 jobs)
NUTS: [a, alpha]
  0%|          | 0/1000 [00:00<?, ?it/s]
```

**Please provide any additional information below.**

## Versions and main components

* PyMC3 Version: 3.4.1
* Theano Version: 
* Python Version: 2.7
* Operating system: OS X
* How did you install PyMC3: (conda/pip) conda

**Comment by junpenglao** [comment]
if it never start sampling and there is no error then it is likely a memory issue. Try setting cores to 1:
`hierarchical_trace = pm.sample(cores=1)` 

You should use softmax instead of logit as logit does not guarantee each row sum to 1. `radon_est = tt.nnet.softmax(a[province_ids])`.

**Comment by aseyboldt** [comment]
It works for me if I replace the logit with a softmax. For a logit I actually get an error message (bad initial energy).

```
radon_est = tt.nnet.softmax(a[province_ids])
radon_like = pm.Multinomial('radon_like', n=observed_data.sum(axis=1), p=radon_est, observed=observed_data)
```

By the way, if the covariance of a MvNormal is a . diagonal matrix, you can just use `pm.Normal` with a shape argument.

@gcgibson I'll close this now, but please reopen if you still have this problem after switching to softmax.
